### PR TITLE
feat: add `cache`option (`options.cache`)

### PR DESCRIPTION
**Pull request by evilebottnawi**
No test here, but tested locally, all works fine :smile:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ module.exports = {
 |**`test`**|`{RegExp\|Array<RegExp>}`|`.`|All assets matching this `{RegExp\|Array<RegExp>}` are processed|
 |**`include`**|`{RegExp\|Array<RegExp>}`|`undefined`|Files to `include`|
 |**`exclude`**|`{RegExp\|Array<RegExp>}`|`undefined`|Files to `exclude`|
+|**`cache`**|`{Boolean\|String}`|`false`|Enable file caching|
```
Comment (michael-ciniawsky) [inline README.md]: Could you do me favor and update the options table with the snippet below (add links to option in README)

```
|**[`name`](#anchor)**|`{Type}`|`<default>`|<description>|
```
```
|Name|Type|Default|Description|
|:--:|:--:|:-----:|:----------|
|**[`test`](#test)**|`{RegExp\|Array<RegExp>}`|`.`|All assets matching this `{RegExp\|Array<RegExp>}` are processed|
|**[`include`](#include)**|`{RegExp\|Array<RegExp>}`|`undefined`|Files to `include`|
|**[`exclude`](#exclude)**|`{RegExp\|Array<RegExp>}`|`undefined`|Files to `exclude`|
|**[`cache`](#cache)**|`{Boolean\|String}`|`false`|Enable file caching|
|**[`asset`](#asset)**|`{String}`|`[path].gz[query]`|The target asset name. `[file]` is replaced with the original asset. `[path]` is replaced with the path of the original asset and `[query]` with the query|
|**[`filename`](#filename)**|`{Function}`|`false`|A `{Function}` `(asset) => asset` which receives the asset name (after processing `asset` option) and returns the new asset name|
|**[`algorithm`](#algorithm)**|`{String\|Function}`|`gzip`|Can be `(buffer, cb) => cb(buffer)` or if a `{String}` is used the algorithm is taken from `zlib`|
|**[`threshold`](#threshold)**|`{Number}`|`0`|Only assets bigger than this size are processed. In bytes.|
|**[`minRatio`](#minratio)**|`{Number}`|`0.8`|Only assets that compress better than this ratio are processed|
|**[`deleteOriginalAssets`](#deleteoriginalassets)**|`{Boolean}`|`false`|Whether to delete the original assets or not|
```

**Comment by evilebottnawi** [comment]
@michael-ciniawsky done. `newFile` => `newAssetName` because it is not asset it is asset name. Also we can rename `asset` option to `assetName`, it is more right and not confusing, but it is breaking change :disappointed:

**Comment by michael-ciniawsky** [comment]
@evilebottnawi Yep, `deleteOriginalAssets` could also be renamed with a better name, but as you said it's sadly a major 😞

**Comment by evilebottnawi** [comment]
@michael-ciniawsky @d3viant0ne we need release :smile:

**Pull request merged**